### ISSUE TITLE
softmax_with_cross_entropy计算多分类任务的损失报错

### ISSUE THREAD

**Issue by LarryLee-BD** [PaddlePaddle/Paddle#22096]
使用fluid.layers.softmax_with_cross_entropy计算多分类任务的损失，label和logits的shape一样，soft_label=True，报错：
_C++ Callstacks:
Enforce failed. Expected logits_dims[i] == labels_dims[i], but received logits_dims[i]:32 != labels_dims[i]:3200.
Input(Logits) and Input(Label) should in same shape in dimensions except axis. at [/paddle/paddle/fluid/operators/softmax_with_cross_entropy_op.cc:134]_

我前置输出label和logit看到shape是一样的
_logits
name: "unstack_0.tmp_0"
type {
  type: LOD_TENSOR
  lod_tensor {
    tensor {
      data_type: FP32
      dims: -1
      dims: 100
    }
  }
}
persistable: false_

_label
name: "read_file_0.tmp_5"
type {
  type: LOD_TENSOR
  lod_tensor {
    tensor {
      data_type: FP32
      dims: -1
      dims: 100
    }
    lod_level: 0
  }
}
persistable: false_

**Comment by gongweibao** [comment]
可能因为数据的问题导致最后shape对不上了。检查一下数据？
